types/http-agent/index.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for http-agent 0.1
+// Project: https://github.com/indexzero/http-agent
+// Definitions by: Ivan Ergunov <https://github.com/hozblok>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import EventEmitter = require('events');
+import request = require('request');
```

Comment:
by default, without anything additional added, you're checking this against current version available on DT, which is `request@2.*`.
The `http-agent` relies on `request@1.*`, is the version of @2 of `request` same, in the context of options, etc?

Reply:
Good question.

I've used `CoreOptions` here. It has a lot of optional fields.

```typescript
    interface CoreOptions {
        baseUrl?: string | undefined;
        callback?: RequestCallback | undefined;
        jar?: CookieJar | boolean | undefined;
        formData?: { [key: string]: any } | undefined;
        form?: { [key: string]: any } | string | undefined;
        auth?: AuthOptions | undefined;
        oauth?: OAuthOptions | undefined;
        aws?: AWSOptions | undefined;
        hawk?: HawkOptions | undefined;
        qs?: any;
        qsStringifyOptions?: any;
        qsParseOptions?: any;
        json?: any;
        jsonReviver?: ((key: string, value: any) => any) | undefined;
        jsonReplacer?: ((key: string, value: any) => any) | undefined;
        multipart?: RequestPart[] | Multipart | undefined;
        agent?: http.Agent | https.Agent | undefined;
        agentOptions?: http.AgentOptions | https.AgentOptions | undefined;
        agentClass?: any;
        forever?: any;
        host?: string | undefined;
        port?: number | undefined;
        method?: string | undefined;
        headers?: Headers | undefined;
        body?: any;
        family?: 4 | 6 | undefined;
        followRedirect?: boolean | ((response: http.IncomingMessage) => boolean) | undefined;
        followAllRedirects?: boolean | undefined;
        followOriginalHttpMethod?: boolean | undefined;
        maxRedirects?: number | undefined;
        removeRefererHeader?: boolean | undefined;
        encoding?: string | null | undefined;
        pool?: PoolOptions | undefined;
        timeout?: number | undefined;
        localAddress?: string | undefined;
        proxy?: any;
        tunnel?: boolean | undefined;
        strictSSL?: boolean | undefined;
        rejectUnauthorized?: boolean | undefined;
        time?: boolean | undefined;
        gzip?: boolean | undefined;
        preambleCRLF?: boolean | undefined;
        postambleCRLF?: boolean | undefined;
        withCredentials?: boolean | undefined;
        key?: Buffer | undefined;
        cert?: Buffer | undefined;
        passphrase?: string | undefined;
        ca?: string | Buffer | string[] | Buffer[] | undefined;
        har?: HttpArchiveRequest | undefined;
        useQuerystring?: boolean | undefined;
    }
```

In fact, only some of the optional fields are used in `http-agent` library. E.g.
https://github.com/indexzero/http-agent/blob/master/lib/http-agent.js#L58
and it's not possible to pass some others fields from `options` (in constructor of `http-agent`) to `request(options)` [here](https://github.com/indexzero/http-agent/blob/master/lib/http-agent.js#L166)

All of those used in `http-agent` (`host`, `headers`, `json`, `encoding` etc) are included to `request@1.*` and `request@2.*`, but there are some additional (e.g. responseBodyStream https://github.com/request/request/blob/v1.2.0/main.js#L70 for `request@1` or `formData` for `request@2`).

I decided to just take the most up-to-date version of the options, realizing that some of them will not be used. Perhaps this is not the best practice and I should only list the ones that the library uses. How do you think? On the other hand, this can be inconvenient, because the user will need to clean out unnecessary fields from the transferred object to the constructor of `http-agent`. And the code of `http-agent` shows that the library was supposed to work with a wider options object, skipping those that are not needed.

Reply:
that's ok, just pointed this to make sure this won't impact usage (e.g. typescript code enforcing options that are not supported at runtime).